modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java

```diff
@@ -96,6 +81,24 @@ public ListTypeDefinition addListTypeDefinition(
 				LocaleUtil.getDefault(), externalReferenceCode));
 	}
 
+	@Indexable(type = IndexableType.REINDEX)
+	@Override
+	public ListTypeDefinition addListTypeDefinition(
+			String externalReferenceCode, long userId,
+			Map<Locale, String> nameMap)
+		throws PortalException {
+
+		_validateName(nameMap, LocaleUtil.getSiteDefault());
+
+		ListTypeDefinition listTypeDefinition =
+			listTypeDefinitionPersistence.create(
+				counterLocalService.increment());
+
+		listTypeDefinition.setExternalReferenceCode(externalReferenceCode);
+
+		return _addListTypeDefinition(userId, listTypeDefinition, nameMap);
```

Comment:
@mateussgn Move `listTypeDefinition.setExternalReferenceCode(externalReferenceCode)` to `_addListTypeDefinition`

Reply:
Sure!
Thanks for the tip!